Salesforce/SalesforceConnector.php

```diff
@@ -209,41 +210,51 @@ private function saveEntitiesToDB(string $intent, $entities, bool $transactional
 
             switch ($intent) {
                 case SalesforceConsumerInterface::CREATED:
+                    $this->logger->info('SalesforceConnector->saveEntitiesToDB()-002 Doing a CREATED persist().');
+                    $manager->persist($entity);
+                    break;
                 case SalesforceConsumerInterface::UPDATED:
                 case SalesforceConsumerInterface::UNDELETED:
-                    $manager->merge($entity);
+                    $this->logger->info('SalesforceConnector->saveEntitiesToDB()-002 Doing an UPDATED persist().');
+                    $manager->persist($entity);
```

Comment:
NOT SO FAST!
You may want to refer to this note on Doctrine concerning [persisting unmanaged entities](https://www.doctrine-project.org/projects/doctrine-orm/en/2.7/reference/working-with-objects.html#persisting-entities)
The purpose of merging an entity on this step is to take entities that have become detached from the EM and to re-sync them before moving on to the flush step.  But when are there unmanaged entities coming through this switch statement you ask?  If you look back through code I am sure you are cognizant that this entity is either unmanaged because it is indeed a new entity, or is managed because our entity locator found it.  In both cases persisting is OK (Although wasteful for entities that are already managed >:C).  What you neglected to do was look __forward__ in code.

On line 269 you may notice this function is called recursively on the event of an error.  An error causes the EM to close which causes all entities to become unmanaged, whether or not they already exist in the database already.  We then try to insert/update the entities that failed one at a time.  merge() was used because we needed a way to [reattach entities](https://github.com/geosocio/entity-attacher/blob/develop/src/EntityAttacher.php) when this happened.  Persist would cause very very bad things to happen.

Reply:
Merge is getting deprecated on the next version of Doctrine ORM; and at first glance, persist looked like the drag-and-drop replacement. My limited testing didn't show an issue, but I'm going to roll this back to merge() to avoid unintended consequences.